cmd/query/app/static_handler.go

```diff
@@ -83,20 +86,23 @@ func NewStaticAssetsHandler(staticAssetsRoot string, options StaticAssetsHandler
 		options.Logger = zap.NewNop()
 	}
 
+	if options.WatcherFactory == nil {
+		options.WatcherFactory = &watcher.FsNotifyWatcherFactory{}
+	}
+
 	indexHTML, err := loadAndEnrichIndexHTML(assetsFS.Open, options)
 	if err != nil {
 		return nil, err
 	}
 
 	h := &StaticAssetsHandler{
-		options:  options,
-		assetsFS: assetsFS,
+		options:        options,
+		assetsFS:       assetsFS,
+		watcherFactory: options.WatcherFactory,
 	}
 
 	h.indexHTML.Store(indexHTML)
-	h.watch()
-
-	return h, nil
+	return h, h.watch()
```

Comment:
Are you sure this is necessary? Failure to instantiate a watcher can now prevent the service from starting.

Reply:
I'm glad you asked as I don't understand how critical UI config watching is and leaned on the side of bubbling errors up.

If it's something that historically has been brittle (as I noticed from unit tests), I'd be happy to revert this back (it'd also simplify the PR); and errors are reported through logging anyway.

Reply:
On the other hand, watching is only enabled (hence possible failure to start) if the user provides a UI configuration and, by default, the UI config path is `""` which means `watch()` should just return a `nil` error. If the user provides UI configuration, they should expect the config to be watched and reloaded automatically. If it fails, I think it should prevent the service from starting, unless if we provide an **opt-out** flag.

If this error is encountered, the error message is quite clear and I believe, in most cases, is a straightforward resolution. e.g.:
```
{"level":"panic","ts":1609905512.493839,"caller":"app/static_handler.go:56","msg":"Could not create static assets handler","error":"cannot read UI config file /does/not/exist.json: open /does/not/exist.json: no such file or directory"
```

Reply:
In case of an obvious error like this I'd agree. But fsnotify/NewWatcher can return an error even without a file name, that error is likely to me much less sensible. Imaging collector is running on an architecture where fsnotify does not work, but the user wasn't interested in hot reload anyway, they just wanted to start with a custom config file. The old code would only log an error and continue working.

Reply:
Okay, agree with this if existing users have an expectation of just running UI config without hot reloads. I'll revert back to returning nil.